compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -952,6 +952,9 @@ fn report_non_exhaustive_match<'p, 'tcx>(
         // Look at the first witness.
         collect_non_exhaustive_tys(cx, &witnesses[0], &mut non_exhaustive_tys);
 
+        // The items are only used to add notes to an err, so the order does not
+        // affect query stability.
+        #[allow(rustc::potential_query_instability)]
         for ty in non_exhaustive_tys {
```

Comment:
The order of diagnostics is observable for the user, so having it platform independent would be better.

Reply:
Using `FxIndexSet` instead of `FxHashSet` for `non_exhaustive_tys` should achieve that goal.